p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -5420,21 +5422,33 @@ the stack in order of increasing indexes.
 ## Data insertion into packets { #sec-deparse-insert }
 
 The ```packet_out``` datatype is defined in the P4 core library, and
-reproduced below. It provides two methods for appending data to an
-output packet; both methods are called ```emit```. The first version
-only accepts headers, and the second one accepts arbitrary data.
+reproduced below. It provides a method for appending data to an
+output packet called ```emit```:
 
 ~ Begin P4Example
 extern packet_out {
-    void emit<T>(in T hdr);
-    void emit<T>(in bool condition, in T data);
+    void emit<T>(in T data);
 }
 ~ End P4Example
 
-The first version of ```emit``` just calls the second one with the
-header validity bit as a condition.
+The ```emit``` method appends the data in its argument, but provides
+special behavior for headers, header stacks, `struct`s, and header
+unions:
+
+- When applied to a header, `emit` appends the data in the header to
+  the packet if it is valid and otherwise behaves like a no-op.
+- When applied to a header stack, `emit` recursively invokes itself to
+  each element of the stack.
+- When applied to a `struct` or header union, `emit` recursively
+  invokes itself to each field.
+
+Appending a value such as a bit-string or integer value to a
```

Comment:
Correct, under the assumptions about T. I've updated the prose and pseudo-code to match.